### PR TITLE
feat(validation): `PipelineValidator` extension point

### PR DESCRIPTION
**Pull request by ajordens**

### CODE REVIEW

File: front50-core/src/main/groovy/com/netflix/spinnaker/front50/validator/GenericValidationErrors.groovy [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+package com.netflix.spinnaker.front50.validator
+
+import org.springframework.validation.AbstractErrors
+import org.springframework.validation.Errors
+import org.springframework.validation.FieldError
+import org.springframework.validation.ObjectError;
+
+class GenericValidationErrors extends AbstractErrors {
```
Comment (ajordens) [inline front50-core/src/main/groovy/com/netflix/spinnaker/front50/validator/GenericValidationErrors.groovy]: This is basically a copy of what we've got in `ApplicationValidationErrors` ... the two can likely be merged in a subsequent PR.

Comment (ajordens) [reply front50-core/src/main/groovy/com/netflix/spinnaker/front50/validator/GenericValidationErrors.groovy]: Not sure it's really all that valuable to use Spring `Errors` but we've got this pattern in a few spots already.

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy [hunk 2]
```diff
@@ -195,6 +203,17 @@ class PipelineController {
...
 
     checkForDuplicatePipeline(pipeline.getApplication(), pipeline.getName().trim(), pipeline.getId())
+
+    def errors = new GenericValidationErrors(pipeline)
+    pipelineValidators.each {
+      it.validate(pipeline, errors)
+    }
+
+    if (errors.hasErrors()) {
+      throw new ValidationException(
```
Comment (ajordens) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy]: There's built-in support already for `ValidationException` in `kork`.

Applications are doing validations with an Application-specific exception, opt'd not to follow that pattern. We can clean that usage up subsequently as well.

Comment (ajordens) [reply front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy]:
```
{
    "timestamp": 1552003316289,
    "status": 400,
    "error": "Bad Request",
    "exception": "com.netflix.spinnaker.kork.web.exceptions.ValidationException",
    "message": "No message available",
    "errors": ["your error message here"]
}
```

Serialized response fits the pattern we've got elsewhere.

File: front50-core/src/main/groovy/com/netflix/spinnaker/front50/validator/GenericValidationErrors.groovy (hunk 1 above)
Comment (robzienert) [reply front50-core/src/main/groovy/com/netflix/spinnaker/front50/validator/GenericValidationErrors.groovy]: I think I'd prefer to use `Errors` over something else, personally. Converge on a standard, and one that we don't have to write 💯

File: front50-core/src/main/groovy/com/netflix/spinnaker/front50/validator/PipelineValidator.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.validator;
+
+import com.netflix.spinnaker.front50.model.pipeline.Pipeline;
+import org.springframework.validation.Errors;
+
+public interface PipelineValidator {
```
Comment (robzienert) [inline front50-core/src/main/groovy/com/netflix/spinnaker/front50/validator/PipelineValidator.java]: I'd like to start seeing javadocs on all interfaces (class- & method-level) we introduce. This one is pretty explanatory, but it'll be a good habit to get into regardless.

Comment (ajordens) [reply front50-core/src/main/groovy/com/netflix/spinnaker/front50/validator/PipelineValidator.java]: :thumbs_up: 

I like and can get behind that.

**Pull request merged**